client/src/pages/ScanStore.tsx

```diff
@@ -1,32 +1,76 @@
 import React, { useEffect, useState } from "react";
 import StoreHeader from "./../components/StoreHeader";
-import Divider from "@material-ui/core/Divider";
-import { Item, emptyItem, CartItem, emptyCartItem } from "./../interfaces";
+import ItemCard from "./../components/ItemCard";
+import Cart from "./../components/Cart";
+import {
+  Fab,
+  Box,
+  Container,
+  Divider,
+  Grid,
+  Paper,
+  Typography,
+} from "@material-ui/core";
+import { spacing } from "@material-ui/system";
+import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
+import EditIcon from "@material-ui/icons/Edit";
+import PaymentIcon from "@material-ui/icons/Payment";
+import AddIcon from "@material-ui/icons/Add";
+import {
+  Item,
+  emptyItem,
+  CartItem,
+  emptyCartItem,
+  MediaResponse,
+  emptyMediaResponse,
+} from "./../interfaces";
 import { fetchJson } from "./../utils";
+import { TextInputField } from "./../components/Components";
+import { BrowserMultiFormatReader } from "@zxing/library";
+import SampleBarcode from "./../img/Sample_EAN8.png";
+declare const window: any;
+
+//Yiheng: For whether we should display our uploaded image for debugging
+const debugImg = true;
 
 function ScanStore() {
   // Update URL params to find storeID
   const curUrl = window.location.search;
   const urlParams = new URLSearchParams(curUrl);
   const storeID = urlParams.get("id");
+  const merchantID = urlParams.get("mid");
 
-  let cartItems: CartItem[] = [];
-
-  // Declare list of items in our cart
+  const [showCart, setShowCart] = useState(false);
   const [shoppingList, setShoppingList] = useState<CartItem[]>([]);
+  //DEBUGGING Current barcode to 'scan'
+  const [curBarcode, setCurBarcode] = useState<string>("");
+
+  // Uploaded image data
+  const [uploadImg, setUploadImg] = useState<MediaResponse>(emptyMediaResponse);
+
+  //TODO: Look into `class ScanStore extends React.Component` syntax
+  //      in order to declare class-level instance variables (cache cart)
+  let cartItems: CartItem[] = [];
+  // Build barcode:idx mapping
+  let idxMap: { [key: string]: number } = {};
 
   // wrapper function to fetch cart items and execute callback upon success
   const fetchCart = async () => {
     let data = {
       "store-id": storeID,
     };
     const cart = await fetchJson(data, "/api/cart");
-    const items = await fetchCartItems(cart)
+    const items = await fetchCartItems(cart);
     updateShoppingList(cartItems, items);
   };
 
   const fetchCartItems = async (items: any) => {
-    cartItems = items; // NI: what is this for?
+    // Empty cart contents
+    cartItems.length = 0;
+    // replace with current items
+    for (let i = 0; i < items.length; ++i) {
```

Comment:
Prefer for-of instead go/js-style#features-for-loops

Reply:
will address while refactoring ScanStore.tsx #30